Assets/Scripts/PositionTracker.cs

```diff
@@ -53,14 +54,10 @@ private void OnTriggerEnter(Collider other)
             Bullseye bullseyeObj = other.GetComponent<Bullseye>();
 
             Rigidbody rb = GetComponent<Rigidbody>();
-            if (rb == null) {
-                Debug.LogError("Racer does not have a rigid body!");
-                return;
-            }
 
             // Fetch collision Point
             RaycastHit hitInfo = new RaycastHit();
-            if(Physics.Raycast(transform.position, rb.velocity, out hitInfo))
+            if(Physics.Raycast(transform.position - rb.velocity, rb.velocity, out hitInfo, (1 << 10)))
```

Comment:
Not sure I understand what's going on here. Mind explaining?

Reply:
`Physics.Raycast(transform.position - rb.velocity, rb.velocity, out hitInfo, (1 << other.gameObject.layer))`

Origin of raycast: `transform.position - rb.velocity`, meaning start the raycast behind the player model
`
(1 << other.gameObject.layer) `is a layer mask which ensures the raycast will only hit objects on the bullseye's layer. Previously I had hard-coded it to layer 10, Bullseye, but technically other.gameObject.layer is better practice because Bullseye's layer might change from 10

For your interest, if you wanted to make a layer mask that hits anything except a given layer, the syntax is `~(1 << x)` where x is the desired layer to ignore

![raycasts](https://user-images.githubusercontent.com/11684949/163685124-65deec64-7146-4de7-a23f-c398a7510a2c.png)